llama_index/vector_stores/google/generativeai/genai_extension.py

```diff
@@ -0,0 +1,528 @@
+        client_options=client_options_lib.ClientOptions(
+            api_endpoint=_config.api_endpoint
+        ),
+    )
+
+
+_default_retriever: genai.RetrieverServiceClient = build_semantic_retriever()
+
+
+def _set_default_retriever(retriever: genai.RetrieverServiceClient) -> None:
+    global _default_retriever
+    _default_retriever = retriever
+
+
+def build_text_service() -> genai.TextServiceClient:
+    return genai.TextServiceClient(
+        client_info=gapic_v1.client_info.ClientInfo(user_agent=_USER_AGENT),
+        client_options=client_options_lib.ClientOptions(
+            api_endpoint=_config.api_endpoint
+        ),
+    )
+
+
+_default_text_service: genai.TextServiceClient = build_text_service()
+
+
+def _set_default_text_service(text_service: genai.TextServiceClient) -> None:
+    global _default_text_service
+    _default_text_service = text_service
+
+
+def list_corpora(
+    *,
+    client: Optional[genai.RetrieverServiceClient] = None,
+) -> Iterator[Corpus]:
+    if client is None:
+        client = _default_retriever
+    for corpus in client.list_corpora(
+        genai.ListCorporaRequest(page_size=_default_page_size)
+    ):
+        yield Corpus.from_corpus(corpus)
+
+
+def get_corpus(
+    *,
+    corpus_id: str,
+    client: Optional[genai.RetrieverServiceClient] = None,
+) -> Optional[Corpus]:
+    if client is None:
+        client = _default_retriever
+    try:
+        corpus = client.get_corpus(
+            genai.GetCorpusRequest(name=str(EntityName(corpus_id=corpus_id)))
+        )
+        return Corpus.from_corpus(corpus)
+    except Exception:
+        return None
+
+
+def create_corpus(
+    *,
+    corpus_id: Optional[str] = None,
+    display_name: Optional[str] = None,
+    client: Optional[genai.RetrieverServiceClient] = None,
+) -> Corpus:
+    if client is None:
+        client = _default_retriever
+
+    name: Optional[str]
+    if corpus_id is not None:
+        name = str(EntityName(corpus_id=corpus_id))
+    else:
+        name = None
+
+    new_display_name = display_name or f"Untitled {datetime.datetime.now()}"
+
+    new_corpus = client.create_corpus(
+        genai.CreateCorpusRequest(
+            corpus=genai.Corpus(name=name, display_name=new_display_name)
+        )
+    )
+
+    return Corpus.from_corpus(new_corpus)
+
+
+def delete_corpus(
+    *,
+    corpus_id: str,
+    client: Optional[genai.RetrieverServiceClient] = None,
+) -> None:
+    if client is None:
+        client = _default_retriever
+    client.delete_corpus(
+        genai.DeleteCorpusRequest(name=str(EntityName(corpus_id=corpus_id)), force=True)
+    )
+
+
+def list_documents(
+    *,
+    corpus_id: str,
+    client: Optional[genai.RetrieverServiceClient] = None,
+) -> Iterator[Document]:
+    if client is None:
+        client = _default_retriever
+    for document in client.list_documents(
+        genai.ListDocumentsRequest(
+            parent=str(EntityName(corpus_id=corpus_id)), page_size=_default_page_size
```

Comment:
Is there any way to specify the page_token so we can get more than the first 20 docs?

Reply:
Another good catch! Just added `page_size` to the `_config` object, which can be set by calling `set_defaults()`.

Reply:
Hmm, I'm not seeing any change. Don't we need to keep track of the next_page_token that comes back from the list_documents response and pass it in to the next call as page_token?